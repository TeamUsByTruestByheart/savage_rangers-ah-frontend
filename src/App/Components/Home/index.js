import React, { Component } from 'react';
import axiosInstance from '../../../../configs/axios';
export class Home extends Component {
  state = {
    up: false,
    down: false
  }
  async componentDidMount() {
    const res = await axiosInstance.get('');
    if (res) this.setState({ up: true })
    else this.setState({ down: true })
  }
  render() {
    if (this.state.up) {
      return (
        <React.Fragment>
          <h1>Authors Heaven</h1>
          <p>This is the home page of authors heaven v 1.0.0</p>
        </React.Fragment>
      )
    } else if (this.state.down) {
      return (
        <React.Fragment>
          <h1>Sorry the server is down</h1>
        </React.Fragment>
      )
    } else {
      setTimeout(() => {
        this.setState({ down: true })
      }, 2000)
      return (
        <React.Fragment>
          <h1>Animation</h1>
        </React.Fragment>
      )

    }
  }
}

export default Home


