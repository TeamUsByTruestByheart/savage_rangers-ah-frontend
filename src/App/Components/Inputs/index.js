import React from 'react';
import PropTypes from 'prop-types';

export const Input = ({
  placeholder, inputMode, onChange, autoComplete, required, type
}) => (
  <input
    className="form-control common-form-input"
    inputMode={inputMode}
    placeholder={placeholder}
    onChange={onChange}
    autoComplete={autoComplete}
    required={required}
    type={type}
  />
);

Input.propTypes = {
  placeholder: PropTypes.string,
  inputMode: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  autoComplete: PropTypes.string,
  required: PropTypes.bool,
  type: PropTypes.string
};

Input.defaultProps = {
  placeholder: '',
  inputMode: 'text',
  autoComplete: 'text',
  required: false,
  type: 'text'
};

export default Input;
