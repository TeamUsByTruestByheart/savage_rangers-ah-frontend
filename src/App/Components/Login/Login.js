import React from 'react';

export default () => {
  return (
    <React.Fragment>
      <h1>Authors Heaven/Login</h1>
      <p>Thi is the login page</p>
    </React.Fragment>
  )
}
