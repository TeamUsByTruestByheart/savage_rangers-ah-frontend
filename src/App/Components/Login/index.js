import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import SimpleReactValidator from 'simple-react-validator';
import propTypes from 'prop-types';
import Input from '../Inputs';
import login from '../../../Redux/Actions/auth';
import Navbar from '../Common/NavProfile/navbar';
import Footer from '../Common/Footer';
import SocialLogin from '../SocialLogin/socialLogin';

import Alert from '../Common/Alert';

export class Login extends Component {
  constructor(props) {
    super(props);
    this.state = { Email: '', Password: '' };
    this.validator = new SimpleReactValidator({ locale: 'en' });
  }

  handleInput = (name, value) => this.setState({ [name]: value });

  handleLogin = (e) => {
    const { dispatchLogin } = this.props;
    const { Email, Password } = this.state;
    e.preventDefault();
    if (this.validator.allValid()) dispatchLogin({ Email, Password });
    else {
      this.validator.showMessages();
      this.forceUpdate();
    }
  };

  render() {
    const { validator, state } = this;

    let errorMessage;
    const {
      authReducer: { errors, isAuthorized, user },
      history
    } = this.props;
    if (!isAuthorized && errors) errorMessage = errors.error || errors.email;
    if (isAuthorized) history.replace(`/profile/${user.username}`);

    return (
      <div className="">
        <Navbar />
        <div className="container custom-login">
          <section className="wrapper">
            <div className="col-center">
              {errorMessage ? <Alert message={errorMessage} type="danger" cssClass="danger" /> : ''}

              <form className="form-horizontal common-form col-sm-6">
                <p className="common-form-h2">Login</p>
                {['Email', 'Password'].map((element, index) => (
                  <div key={String(index)} style={{ width: '100%' }} className="form-group">
                    <Input
                      name="input"
                      className="form-control col-sm-12"
                      placeholder={element}
                      inputMode={element}
                      onChange={({ target }) => this.handleInput(element, target.value)}
                      autoComplete={element}
                      required
                      type={element}
                    />

                    {index === 0
                      ? validator.message('email', state.Email, 'required|email', { className: 'text-danger' })
                      : validator.message('password', state.Password, 'required|string|min:8', { className: 'text-danger' })}
                  </div>
                ))}
                <div className="links">
                  <p className="float-left m-3 links links-right">
                    Don't have an account ?
                    {' '}
                    <a href="/signup">Sign up</a>
                  </p>
                  <Link className="float-right m-3 links links-right" to="/forgot-password">
                    Forgot your password?
                  </Link>
                </div>
                <button
                  id="login-button"
                  type="submit"
                  name="submit"
                  className="btn btn-block col-sm-8 common-form-button"
                  onClick={this.handleLogin}
                >
                  Login
                </button>
                <SocialLogin />
              </form>
            </div>
          </section>
        </div>
        <Footer />
      </div>
    );
  }
}

Login.propTypes = {
  dispatchLogin: propTypes.func.isRequired,
  authReducer: propTypes.object,
  history: propTypes.object
};

export const mapStateToProps = ({ authReducer }) => ({ authReducer });

export const mapDispatchToProps = dispatch => ({ dispatchLogin: user => dispatch(login(user)) });

export default connect(mapStateToProps,
  mapDispatchToProps)(Login);
