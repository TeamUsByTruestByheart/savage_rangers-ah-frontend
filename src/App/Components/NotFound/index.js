
import React from 'react';
import TextBox from './TextBox';


export default () => {
  return (
    <React.Fragment>
      <h1>Authors Heaven</h1>
      <TextBox />
    </React.Fragment>
  )
}
