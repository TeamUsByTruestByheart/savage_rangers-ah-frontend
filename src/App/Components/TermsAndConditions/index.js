import React, { Component } from 'react';
import Navbar from '../Common/NavProfile/navbar';
import Footer from '../Common/Footer';

export class TermsAndConditions extends Component {
  render() {
    return (
      <div className="">
        <Navbar />
        <div className="container .custom-footer">
          <section className="wrapper" />

        </div>
        <Footer />
      </div>
    );
  }
}
export default TermsAndConditions;
