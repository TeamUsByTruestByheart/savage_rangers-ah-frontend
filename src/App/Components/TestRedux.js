import React, { Component } from 'react';
import propTypes from 'prop-types';
import { connect } from 'react-redux';
import testReduxAction from '../../Redux/Actions/testReduxActions';

const { getWelcome } = testReduxAction;
/**
 *
 *This class is just for showing how we will use Redux,
 * Once the home page will be done, will no longer be needed
 * @export
 * @class TestRedux
 * @extends {Component}
 */
export class TestRedux extends Component {
  componentDidMount() {
    this.props.getWelcome();
  }

  render() {
    return (
      <div>
        <p>
          message loaded from redux:
          {this.props.data ? this.props.data.message : ''}
        </p>
      </div>
    );
  }
}

TestRedux.propTypes = { getWelcome: propTypes.func.isRequired };

// get data from redux and put it in props
export const mapStateToProps = state => ({ data: state.testRedux.data });

// export the component while connecting to redux
export default connect(mapStateToProps,
  { getWelcome })(TestRedux);
