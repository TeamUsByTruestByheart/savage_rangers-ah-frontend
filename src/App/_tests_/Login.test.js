import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { mount, shallow } from '../../enzyme';
import { Login, mapDispatchToProps, mapStateToProps } from '../Components/Login';

const props = {
  history: { replace: jest.fn(), push: jest.fn() },
  handleLogin: jest.fn(),
  dispatchLogin: jest.fn(),
  authReducer: {
    user: { token: 'token', username: 'alain' },
    errors: {},
    isAuthorized: true
  }
};

const shallowed = shallow(<Login {...props} />);
shallowed.setProps({
  authReducer: {
    user: { token: 'token', username: 'alain' },
    errors: { email: 'confirm your email first' }
  }
});

const instance = shallowed.instance();

describe('Login', () => {
  const renderLogin = (args) => {
    const defaultProps = {
      dispatchLogin: jest.fn(),
      authReducer: {
        user: { token: 'token', username: 'alain' },
        errors: {},
        isAuthorized: true
      },
      history: { replace: jest.fn() }
    };
    const final = { ...defaultProps, ...args };
    const login = mount(<MemoryRouter>
      <Login {...final} />
                        </MemoryRouter>);
    return login;
  };
  it('should pop up an alert when there is an error', () => {
    const prop = {
      handleLogin: jest.fn(),
      authReducer: {
        user: { token: 'token', username: 'alain' },
        errors: { error: 'invalid message' },
        isAuthorized: false
      },
      dispatchLogin: jest.fn()
    };
    shallow(<Login {...prop} />);
  });
    it('should map the state to props', () => {
      mapStateToProps({ authReducer: {} });
    });
    it('should map the state to props', () => {
      mapDispatchToProps(jest.fn()).dispatchLogin({ Email: 'dhjv', Password: 'dskxfbjhk' });
    });
    it('should handle input change', () => {
      const handleInput = jest.spyOn(instance, 'handleInput');
      const input = shallowed.find('[name="input"]').at(0);
      const event = { target: { name: 'Email', value: 'diane@gmail.com' } };
      input.simulate('change', event);
      expect(handleInput).toHaveBeenCalled();
    });
    it('should throw an error since data is invalid', () => {
      instance.setState({ Email: '', Password: '' });
      jest.mock('simple-react-validator');
      jest.spyOn(instance, 'handleLogin');
      const btn = shallowed.find('[name="submit"]');
      const event = { preventDefault: jest.fn() };
      btn.simulate('click', event);
    });
    it('should handle data submission', () => {
      instance.setState({ Email: 'diane@gmail.com', Password: 'murekateteDiane' });
      jest.mock('simple-react-validator');
      jest.spyOn(instance, 'handleLogin');
      const btn = shallowed.find('[name="submit"]');
      const event = { preventDefault: jest.fn() };
      btn.simulate('click', event);
    });
  describe('Login', () => {
        it('renders the login component', () => {
          const wrapper = renderLogin();
          wrapper.setState({ Email: '', Password: '' });
          expect(wrapper).toMatchSnapshot();
        });
        it('renders the login form', () => {
          const wrapper = renderLogin();
          expect(wrapper.find('form').length).toBe(1);
        });
    it('shows error message when login fails', () => {
      const authReducer = {
        errors: { error: 'Invalid email or password' },
        isAuthorized: false
      };
      const wrapper = renderLogin();
      wrapper.setProps({ authReducer });
    });
  });
});
