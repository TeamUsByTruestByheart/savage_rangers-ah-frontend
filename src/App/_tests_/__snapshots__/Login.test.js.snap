// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Login renders the login component 1`] = `ReactWrapper {}`;
